src/main/java/org/jenkinsci/plugins/xunit/XUnitProcessor.java

```diff
@@ -144,24 +152,47 @@ public XUnitProcessor(@Nonnull TestType[] tools,
 
     public void process(Run<?, ?> build, FilePath workspace, TaskListener listener, Launcher launcher, @Nonnull Collection<TestDataPublisher> testDataPublishers)
      throws IOException, InterruptedException {
+        TestResult testResult = processXunit(build, workspace, launcher, listener, testDataPublishers, null);
+
+        if (testResult != null) {
+            processDeletion(workspace);
```

Comment:
Performed twice, anyway I prefer to have only a method instead of two method on processor

Reply:
Whoops, I meant to move that out of the common method. Fixing.